src/test/fuzz/process_message_e2e.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2020-2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <consensus/consensus.h>
+#include <net.h>
+#include <net_processing.h>
+#include <node/connection_types.h>
+#include <primitives/transaction.h>
+#include <protocol.h>
+#include <script/script.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/util/mining.h>
+#include <test/util/setup_common.h>
+#include <util/chaintype.h>
+#include <util/check.h>
+#include <util/time.h>
+#include <validationinterface.h>
+
+#include <algorithm>
+#include <cstdlib>
+#include <memory>
+#include <optional>
+#include <stdint.h>
+#include <string.h>
+#include <string>
+#include <string_view>
+#include <vector>
+
+// There are two sets of tests:
+// - direct: pass the fuzzed message directly to PeerManager::ProcessMessage() (fuzz/process_message.cpp)
+// - e2e: send the fuzzed message to a mocked socket which is read from Connman
+//   and processed by PeerManager (like it is happening in the "real world") (fuzz/process_message_e2e.cpp)
+
+namespace {
+NetTestingSetup* g_setup;
+std::string_view LIMIT_TO_MESSAGE_TYPE{};
+} // namespace
+
+void initialize_process_message_e2e()
+{
+    if (const auto val{std::getenv("LIMIT_TO_MESSAGE_TYPE")}) {
+        LIMIT_TO_MESSAGE_TYPE = val;
+        Assert(std::count(getAllNetMessageTypes().begin(), getAllNetMessageTypes().end(), LIMIT_TO_MESSAGE_TYPE)); // Unknown message type passed
+    }
+
+    static const auto testing_setup = MakeNoLogFileContext<NetTestingSetup>(
+            /*chain_type=*/ChainType::REGTEST,
+            /*extra_args=*/{"-txreconciliation"});
+    g_setup = testing_setup.get();
+    for (int i = 0; i < 2 * COINBASE_MATURITY; ++i) {
+        MineBlock(g_setup->m_node, CScript() << OP_TRUE);
+    }
+    SyncWithValidationInterfaceQueue();
+}
+
+FUZZ_TARGET(process_message_e2e, .init = initialize_process_message_e2e)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+
+    const std::string random_message_type{fuzzed_data_provider.ConsumeBytesAsString(CMessageHeader::COMMAND_SIZE).c_str()};
+    if (!LIMIT_TO_MESSAGE_TYPE.empty() && random_message_type != LIMIT_TO_MESSAGE_TYPE) {
+        return;
+    }
+
+    // any time to successfully reset ibd; use a time in the future because Connman is already running
+    SetMockTime(GetMockTime() + std::chrono::minutes{fuzzed_data_provider.ConsumeIntegral<uint8_t>()});
```

Comment:
I stared at this a bit and realized it does not make sense without the `ResetIbd()` call and further, the point is to jump out of IBD at some point later during the test (by changing the mock time so that it is near or before the chain tip). Added that and some comments.